Azure.ResourceManager.Core/ResourceGroupOperations.cs

```diff
@@ -232,11 +257,15 @@ public async Task<ArmOperation<ResourceGroup>> StartAddTagAsync(string name, str
         /// <typeparam name="TOperations"> The type of the operations class for a specific resource. </typeparam>
         /// <typeparam name="TResource"> The type of the class containing properties for the underlying resource. </typeparam>
         /// <returns> Returns a response with the <see cref="ArmResponse{TOperations}"/> operation for this resource. </returns>
+        /// <exception cref="ArgumentException"> Name cannot be null. </exception>
         public ArmResponse<TOperations> CreateResource<TContainer, TOperations, TResource>(string name, TResource model, LocationData location = default)
             where TResource : TrackedResource
             where TOperations : ResourceOperationsBase<TOperations>
             where TContainer : ResourceContainerBase<TOperations, TResource>
         {
+            if (string.IsNullOrWhiteSpace(name))
+                throw new ArgumentException(nameof(name), $"{nameof(name)} provided cannot be null or a whitespace.");
+
             var myResource = model as TrackedResource;
 
             if (myResource == null)
```

Comment:
do we have a test case that exercises this path?  I would imagine this would fail if this was ever true.

Also use (myResource is null)